### PR TITLE
Add initial IngestAPI service

### PR DESCRIPTION
**Pull request by vimto**
### Summary of changes
- A new Spring Boot project for the IngestAPI service, based on the existing Manager service
- A Bean to auto-create the water.allocation Kafka Topic
- A HTTP Endpoint to receive a JSON list of water allocation data
- A Kafka producer, which sends this allocation data to the Kafka water.allocation topic
- Spring security config to protect this endpoint with HTTP Basic Auth
- A custom UserDetailsService to read credentials for Basic auth from a JSON serialized ENV variable (populated by AWS Secrets Manager)
- CI Config for building, testing and deploying the service to ECR
- Batect tasks/containers to run Kafka locally for development
- A Consumer which simply logs messages from the water.allocation topic for debugging

### Things left to consider and card
- Tests! Would be great to pair on these @smozely 
- Should the EOPIngestAPIConfig use different keystore values to the Manager?
- Should we be encrypting API tokens before storing them in AWS Secrets Manager?
- Is the config for the water.allocation topic appropriate?
- What's our approach for naming/name-spacing of API endpoints and Kafka Topics?
- Likewise, what's our approach for Topic granularity in relation to different message types?
- Do we need to update how we format logs so we can query them by logging contexts, such as [ingestId](https://github.com/Greater-Wellington-Regional-Council/Environmental-Outcomes-Platform/pull/29/files#diff-f1032bcb950c3857ebae8e66e2d33058b5d4a0b8dfc4ce4800b5e3c962e2f3cbR32-R36)?
- Once we've finalised the above, we need to add a Readme

### CODE REVIEW

**Review by smozely — COMMENTED** [review]
The `batect-0.82.0.jar` file shouldn't be in the repo should it?

**Comment by vimto** [comment]
> The `batect-0.82.0.jar` file shouldn't be in the repo should it?

No good catch. Have [removed](3cc914f9a14b94d300a773880d6622172a1a9228).

File: packages/IngestAPI/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
```
Comment (vimto) [inline packages/IngestAPI/gradle/wrapper/gradle-wrapper.properties:5]: @smozely given your comment re. the Batect.jar, should these files be committed? I notice they are for the Manager service.

**Comment by smozely** [comment]
> What's our approach for naming/name-spacing of API endpoints and Kafka Topics?
Good question, worth a bit of discussion

> Likewise, what's our approach for Topic granularity in relation to different message types?
I believe its generally encouraged for one message per topic, in particular the Schema registry works that way

> Do we need to update how we format logs so we can query them by logging contexts, such as ingestId
Yes that would be good, its also useful to do in the Web Request context

**Pull request merged**